reference/opt/shaders-msl/vert/layer.msl11.invalid.vert

```diff
@@ -0,0 +1,24 @@
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using namespace metal;
+
+struct main0_out
+{
+    float4 gl_Position [[position]];
+    uint gl_Layer [[render_target_array_index]];
+};
+
+struct main0_in
+{
+    float4 coord [[attribute(0)]];
+};
+
+vertex main0_out main0(main0_in in [[stage_in]])
+{
+    main0_out out = {};
+    out.gl_Position = in.coord;
+    out.gl_Layer = int(in.coord.z);
```

Comment:
Honestly, I didn't think it was a problem, since MSL handles implicit `int`<->`uint` conversions OK. (The loads might be a problem, though. I wonder if we need to add yet more cases to `bitcast_from_builtin_load` for this reason.)

Reply:
The biggest concern is probably arith shift vs logical shift in case the shader is doing something really weird, but better to be safe than sorry here I think.